pkg/controller/common/defaults/pod_template_test.go

```diff
@@ -1122,7 +1122,7 @@ func TestPodTemplateBuilder_WithPreStopHook(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			b := NewPodTemplateBuilder(tt.podTemplate, "mycontainer")
-			got := b.WithPreStopHook(tt.preStopHook).Container.Lifecycle
+			got := b.WithPreStopHook(tt.preStopHook).containerDefaulter.Container().Lifecycle
 			if !reflect.DeepEqual(got.PreStop, &tt.wantPreStop) {
 				t.Errorf("PreStop after PodTemplateBuilder.WithPreStopHook() = %v, want %v", got.PreStop, tt.wantPreStop)
 			}
```

Comment:
Should most of those unit tests live in `common/defaults/defaulter_test.go` now?

Reply:
The `Defaulter` is tested through the use of some `PodTemplate`, I guess it's fine to let them here ?

Reply:
Yeah I'm fine with that.